nixos/modules/services/games/asf.nix

```diff
@@ -0,0 +1,203 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.archisteamfarm;
+
+  format = pkgs.formats.json { };
+
+  asf-config = format.generate "ASF.json" (cfg.config // {
+    # we disable it because ASF cannot update itself anyways
+    # and nixos takes care of restarting the service
+    # is in theory not needed as this is already the default for default builds
+    UpdateChannel = 0;
+  });
+
+  mkBot = n: c:
+    builtins.toJSON (c.settings // {
+      SteamLogin = if c.username == "" then n else c.username;
+      SteamPassword = c.passwordFile;
+      # sets the password format to file (https://github.com/JustArchiNET/ArchiSteamFarm/wiki/Security#file)
+      PasswordFormat = 4;
+      Enabled = c.enabled;
+      name = n;
+    });
+in
+{
+  options.services.archisteamfarm = {
+    enable = mkOption {
+      type = types.bool;
+      description = ''
+        If enabled, starts the Archis Steam Farm service.
+        For configuring the SteamGuard token you will need to use the web-ui, which is enabled by default over on 127.0.0.1:1242.
+        You cannot configure ASF in any way outside of nix, since all the config files get wiped on restart and replaced with the programatically set ones by nix.
+      '';
+      default = false;
+    };
+
+    web-ui = mkOption {
+      type = types.submodule {
+        options = {
+          enable = mkEnableOption
+            "Wheter to start the web-ui. This is the preffered way of configuring things such as the steam guard token";
+
+          package = mkOption {
+            type = types.package;
+            default = pkgs.ASF-ui;
+            description =
+              "Web ui package to use. Contents must be in $out/lib/dist.";
```

Comment:
```suggestion
              "Web ui package to use. Contents must be in $''{pkgs.ASF-ui }/lib/dist.";
```

Reply:
This doesn't work, did you mean `''${...}`?
But I also find this confusing, since we have `cfg.web-ui.package` and a description that would hardcode ASF-ui would give the impression we don't listen to the option

Reply:
> This doesn't work, did you mean ''${...}?

Yeah. Didn't remember how to do it properly.

>  But I also find this confusing, since we have cfg.web-ui.package and a description that would hardcode ASF-ui would give the impression we don't respect the option

Maybe set it to the config.... then? $out is definitely wrong here-

Reply:
I'll just leave away the $out part:

> Contents must be in lib/dist

should be clear enough.